### PR TITLE
Add ciem-aws-iam-overly-permissive-github-role

### PR DESCRIPTION
**Pull request by aidansteele**
Thanks for making CNAPPgoat - it looks pretty promising. In order to teach myself how it works, I tried creating a scenario. I had a fair bit of confusion while doing so, so I won't be surprised (or offended) if this is not remotely sufficient quality to be merged. (I'll also create a separate issue describing the difficulties I had with contributing)

The idea here is that missing conditions on the `sub` claim in the Github.com JWT is an increasingly common misconfiguration. Here's the latest article about the UK govt making this mistake on the [Datadog security blog](https://securitylabs.datadoghq.com/articles/exploring-github-to-aws-keyless-authentication-flaws/). The blog links to other related research.

One potential gotcha is that AWS IAM only allows a single OIDC IDP object to be created per URL. So if an AWS account already has the Github OIDC IDP defined, this scenario fails to deploy. Maybe there's a way to conditionally create the IDP in that case, but I've only known Pulumi for 15 minutes, so it's beyond my skills here.

I don't know what CIEM, CSPM etc is so I copied the `ciem-aws-iam-external-id-3rd-party-role` scenario as template because it seemed most similar.

### CODE REVIEW

File: scenarios/ciem/aws/iam-overly-permissive-github-role/Pulumi.yaml
Comment (noamsdahan) [inline scenarios/ciem/aws/iam-overly-permissive-github-role/Pulumi.yaml:1]: move to scenario/ciem/aws/iam-github-role-without-sub-condition please

File: scenarios/ciem/aws/iam-overly-permissive-github-role/Pulumi.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: ciem-aws-iam-overly-permissive-github-role
```
Comment (noamsdahan) [inline scenarios/ciem/aws/iam-overly-permissive-github-role/Pulumi.yaml:1]:
```suggestion
name: ciem-aws-iam-github-role-without-sub-condition
```

File: scenarios/ciem/aws/iam-overly-permissive-github-role/go.mod [hunk 2]
```diff
@@ -0,0 +1,71 @@
+module iam-external-id-3rd-party-role
```
Comment (noamsdahan) [inline scenarios/ciem/aws/iam-overly-permissive-github-role/go.mod:1]:
```suggestion
module iam-github-role-without-sub-condition
```

File: scenarios/ciem/aws/iam-overly-permissive-github-role/Pulumi.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  "token.actions.githubusercontent.com:sub" in your IAM role trust policy during 
+  the creation process.
+cnappgoat-params:
+  description: This scenario creates an IAM OpenID Connect provider for Github.com
+    and a role that trusts it without sufficient trust policy conditions. This exposes
+    your account to confused deputy attacks. To fix, include the condition
+    "token.actions.githubusercontent.com:sub" in your IAM role trust policy during
+    the creation process.
+  friendlyName: Federated Github.com IAM Role Without Subject Condition
+  id: ciem-aws-iam-overly-permissive-github-role
```
Comment (noamsdahan) [inline scenarios/ciem/aws/iam-overly-permissive-github-role/Pulumi.yaml:15]:
```suggestion
  id: ciem-aws-iam-github-role-without-sub-condition
```

File: scenarios/ciem/aws/iam-overly-permissive-github-role/main.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+			},
+			Description: pulumi.String("This is a vulnerable role that allows any repo on Github.com to assume it"),
+			Tags: pulumi.StringMap{
+				"Cnappgoat": pulumi.String("true"),
+			},
+		})
+		if err != nil {
+			return err
+		}
+		ctx.Export("roleName", role.Name)
```
Comment (noamsdahan) [inline scenarios/ciem/aws/iam-overly-permissive-github-role/main.go:52]:
```suggestion
		ctx.Export("roleArn", role.Arn)
```

**Review by noamsdahan — CHANGES_REQUESTED** [review]
Looks great overall! I prefer the name `ciem-aws-iam-github-role-without-sub-condition` with module and directory renamed accordingly. The Pulumi itself looks good.

**Comment by noamsdahan** [comment]
I'm looking into using an existing OIDC provider if it exists

File: scenarios/ciem/aws/iam-overly-permissive-github-role/main.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/iam"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		idp, err := iam.NewOpenIdConnectProvider(ctx, "CnappGoatGithubOidcProvider", &iam.OpenIdConnectProviderArgs{
```
Comment (noamsdahan) [inline scenarios/ciem/aws/iam-overly-permissive-github-role/main.go:10]:
```suggestion
		oidcUrl := "https://token.actions.githubusercontent.com"
		var idpArn pulumi.StringOutput

		getIdpResult, err := iam.GetOpenidConnectProvider(ctx, &iam.GetOpenidConnectProviderArgs{
			Url: pulumi.StringRef(oidcUrl),
		}, nil)
		if err == nil {
			idpArn = pulumi.String(getIdpResult.Arn).ToStringOutput()
		} else {
		idp, err := iam.NewOpenIdConnectProvider(ctx, "CnappGoatGithubOidcProvider", &iam.OpenIdConnectProviderArgs{
```

File: scenarios/ciem/aws/iam-overly-permissive-github-role/main.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+				"Cnappgoat": pulumi.String("true"),
+			},
+			ThumbprintLists: pulumi.StringArray{
+				pulumi.String("6938fd4d98bab03faadb97b34396831e3780aea1"),
+				pulumi.String("1c58a3a8518e8759bf075b76b750d4f2df264fcd"),
+			},
+		})
+		if err != nil {
+			return err
+		}
```
Comment (noamsdahan) [inline scenarios/ciem/aws/iam-overly-permissive-github-role/main.go:25]:
```suggestion
		}
		
			idpArn = idp.Arn
		}
		
```

**Review by noamsdahan — CHANGES_REQUESTED** [review]
If I put everything in the right place this should make the scenario work with a pre-existing OIDC provider, full code should look like this:

```
package main

import (
	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/iam"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		oidcUrl := "https://token.actions.githubusercontent.com"
		var idpArn pulumi.StringOutput

		getIdpResult, err := iam.GetOpenidConnectProvider(ctx, &iam.GetOpenidConnectProviderArgs{
			Url: pulumi.StringRef(oidcUrl),
		}, nil)
		if err == nil {
			idpArn = pulumi.String(getIdpResult.Arn).ToStringOutput()
		} else {
			// If the OIDC provider doesn't exist, create a new one
			idp, err := iam.NewOpenIdConnectProvider(ctx, "CnappGoatGithubOidcProvider", &iam.OpenIdConnectProviderArgs{
				Url: pulumi.String(oidcUrl),
				ClientIdLists: pulumi.StringArray{
					pulumi.String("sts.amazonaws.com"),
				},
				Tags: pulumi.StringMap{
					"Cnappgoat": pulumi.String("true"),
				},
				ThumbprintLists: pulumi.StringArray{
					pulumi.String("6938fd4d98bab03faadb97b34396831e3780aea1"),
					pulumi.String("1c58a3a8518e8759bf075b76b750d4f2df264fcd"),
				},
			})
			if err != nil {
				return err
			}
			idpArn = idp.Arn
		}

		role, err := iam.NewRole(ctx, "CnappGoatOverlyPermissionGithubRole", &iam.RoleArgs{
			AssumeRolePolicy: pulumi.Sprintf(`{
				"Version": "2012-10-17",
				"Statement": [
					{
						"Sid": "",
						"Effect": "Allow",		
						"Principal": {		
							"Federated": "%s"	
						},		
						"Action": "sts:AssumeRoleWithWebIdentity"
					}
				]
			}`, idpArn),
			ManagedPolicyArns: pulumi.StringArray{
				pulumi.String("arn:aws:iam::aws:policy/AWSDenyAll"),
			},
			Description: pulumi.String("This is a vulnerable role that allows any repo on Github.com to assume it"),
			Tags: pulumi.StringMap{
				"Cnappgoat": pulumi.String("true"),
			},
		})
		if err != nil {
			return err
		}
		ctx.Export("roleArn", role.Arn)
		return nil
	})
}
```

**Comment by aidansteele** [comment]
Hi folks,

I really appreciate the detailed feedback you provided, but unfortunately I won't have the bandwidth to work on this for a while (even though it's pretty small!) - so I'll close this PR for now, rather than keep it in a limbo state. For the record, I'm totally happy for you or anyone else to resubmit an identical scenario.

**Pull request closed without merging**